packages/network/src/createTxQueue.ts

```diff
@@ -219,6 +224,8 @@ export function createTxQueue<C extends Contracts>(
         // Nonce handeling
         if (shouldIncreaseNonce) incNonce();
         if (shouldResetNonce) await resetNonce();
+        // Bubble up error
+        if (error) txRequest.cancel(error);
```

Comment:
This still doesn't help with tx errors if it passes the gas estimation and returns a plain "transaction failed" error message:
<img width="412" alt="image" src="https://user-images.githubusercontent.com/508855/213322244-388171a6-2712-45b3-9e98-8dffc878f9da.png">